### PR TITLE
TW-1169 share multiple files external

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket

- #1169 

#### iOS

https://github.com/linagora/twake-on-matrix/assets/99852347/c11601fc-b758-4914-9322-23f34e41dcfb

https://github.com/linagora/twake-on-matrix/assets/99852347/54be03ab-009e-4f7a-b8b0-746c887649f8

https://github.com/linagora/twake-on-matrix/assets/99852347/c69d67ec-792b-4239-af64-6a0cd30363f1

#### Android

https://github.com/linagora/twake-on-matrix/assets/99852347/728fcbe4-a6ea-457d-a79d-f78e29b8f41d

https://github.com/linagora/twake-on-matrix/assets/99852347/333fde99-9f68-4122-be61-57c3ac4d7e26

https://github.com/linagora/twake-on-matrix/assets/99852347/0e24fa0b-cab5-4952-85bd-9dc7855484a8

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -82,6 +82,37 @@
...
+                <data android:mimeType="video/*" />
+            </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.SEND" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="*/*" />
+            </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.SEND_MULTIPLE" />
+                <category android:name="android.intent.category.DEFAULT" />
```
Comment (sherlockvn) [inline android/app/src/main/AndroidManifest.xml:115]: mimeType = */* is enough and include video, images, ...

File: lib/pages/chat/chat.dart [hunk 2]
```diff
@@ -375,7 +378,9 @@ class ChatController extends State<Chat>
         setReadMarker();
         return;
       }
-      _initUnreadLocation(fullyRead);
+      if (room?.hasNewMessages == true) {
+        _initUnreadLocation(fullyRead);
+      }
```
Comment (sherlockvn) [inline lib/pages/chat/chat.dart:383]: why we need this change ?

File: lib/pages/share/share.dart [hunk 3]
```diff
@@ -84,23 +84,52 @@ class ShareController extends State<Share>
       id: selectedRoomsNotifier.value.first,
       client: Matrix.of(context).client,
     );
+    final shareContentList = Matrix.of(context).shareContentList;
     final shareContent = Matrix.of(context).shareContent;
```
Comment (sherlockvn) [inline lib/pages/share/share.dart:88]: remove `shareContent` if not needed, i see you remove it in `receive_sharing_intent_mixin.dart` already

File: lib/pages/chat/chat.dart (hunk 2 above)
Comment (nqhhdev) [reply lib/pages/chat/chat.dart:383]: If don't have a new message, always call `_initUnreadLocation`  and jump to the last read message. 
So we need to check if hasNewMessgae => call `_initUnreadLocation `

File: lib/pages/share/share.dart (hunk 3 above)
Comment (nqhhdev) [reply lib/pages/share/share.dart:88]: `shareContent` support to text, keep it. Only change for sharing files.

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**